ReactAndroid/src/main/java/com/facebook/react/bridge/JavaOnlyArray.java

```diff
@@ -60,12 +88,12 @@ public boolean isNull(int index) {
 
   @Override
   public double getDouble(int index) {
-    return (Double) mBackingList.get(index);
```

Comment:
Were these broken before?

Reply:
So there was a small inconsistency with the way getDouble and getInt work with JavaOnlyArray or Map vs NativeArray or Map. The problem was that in JS we only have "numbers" so both int and double map to the same type. Therefore NativeArray or Map is also casting the value to an appropriate type at the moment when getDouble or getInt is called. So even if we send a floating point from the JS, java code can call getInt and it will return a rounded integer. Same if we send an integer from JS. 

This has became a problem since in animated driver tracking node I need to clone NativeArray/Map to JavaOnlyArray/Map, and when there are fields of type "Number" I can't tell if these are floating point numbers or not. Because of that I always read and write them as double. Then when I pass that JavaOnlyMap/Array down to the driver it knows about semantics of the fields and thus it may call getInt on some fields. In that case it was failing because the number has been set as a double.